src/components/user-activity.ts

```diff
@@ -116,14 +120,17 @@ export class UserActivityTracker {
         }
 
         this.dataSet.users[userId] = userObject;
-        setImmediate(() => {
-            log.debug("Notifying the listener of RMAU changes");
-            this.onChanges?.({
-                changed: [userId],
-                dataSet: this.dataSet,
-                activeUsers: this.countActiveUsers().allUsers,
-            });
-        });
+        if (!this.debounceTimer) {
+            this.debounceTimer = setTimeout(() => {
+                log.debug("Notifying the listener of RMAU changes");
```

Comment:
```suggestion
                log.debug(`Notifying the listener of RMAU changes for ${Object.keys(this.dataSet).length} users`);
```

Though, I wonder if we should make `users` a Map due to:

- prototype pollution scares
- giving us a `.size` property?